src/charm.py

```diff
@@ -576,38 +636,26 @@ def _api_address(self) -> str:
         """
         return f"https://{self._bind_address}:{self.VAULT_PORT}"
 
-    def _push_certificates_to_workload(
-        self,
-        private_key: str,
-        certificate: str,
-        ca_certificate: str,
-    ) -> None:
-        """Push the certificates to the workload."""
-        self._container.push(path=TLS_CERT_FILE_PATH, source=certificate)
+    def _push_ca_certificate_to_workload(self, certificate: str) -> None:
+        """Push the CA certificate to the workload."""
+        self._container.push(path=TLS_CA_FILE_PATH, source=certificate)
+        logger.info("Pushed CA certificate to workload")
+
+    def _push_unit_certificate_to_workload(self, private_key: str, certificate: str) -> None:
+        """Push the unit certificate to the workload."""
         self._container.push(path=TLS_KEY_FILE_PATH, source=private_key)
-        self._container.push(path=TLS_CA_FILE_PATH, source=ca_certificate)
-        logger.info("Pushed certificates to workload")
-
-    def _certificate_pushed_to_workload(
-        self, private_key: str, certificate: str, ca_certificate: str
-    ) -> bool:
-        """Check if the certificates are pushed to the workload."""
-        if not self._container.exists(path=TLS_CERT_FILE_PATH):
-            return False
-        if not self._container.exists(path=TLS_KEY_FILE_PATH):
-            return False
-        if not self._container.exists(path=TLS_CA_FILE_PATH):
-            return False
-        existing_certificate = self._container.pull(path=TLS_CERT_FILE_PATH)
-        if existing_certificate.read() != certificate:
-            return False
-        existing_private_key = self._container.pull(path=TLS_KEY_FILE_PATH)
-        if existing_private_key.read() != private_key:
-            return False
-        existing_ca_certificate = self._container.pull(path=TLS_CA_FILE_PATH)
-        if existing_ca_certificate.read() != ca_certificate:
-            return False
-        return True
+        self._container.push(path=TLS_CERT_FILE_PATH, source=certificate)
+        logger.info("Pushed unit certificate to workload")
+
+    def _ca_certificate_pushed_to_workload(self) -> bool:
+        """Check if the CA certificate is pushed to the workload."""
+        return self._container.exists(path=TLS_CA_FILE_PATH)
+
+    def _unit_certificate_pushed_to_workload(self) -> bool:
```

Comment:
Docstring arrangement as above

Reply:
I improved the docstring but ultimately I don't agree that we should add return docstring in such obvious cases. That would be just for the sake of it and not make the code more readable.